lib/Project.js

```diff
@@ -455,6 +455,33 @@ Project.prototype.close = function(cb) {
     var base = this.options.id;
     var extractedPath = path.join(dir, base + "-extracted.xliff");
     var extracted = new TranslationSet(this.sourceLocale);
+    var resourceList = [];
+
+    if (typeof this.options.manifest !== undefined) {
+        logger.info(" Creating manifest file ...");
+        var resourcePath = this.options.manifest["dir"] || ".";
+        var manifestName = this.options.manifest["name"] || "ilibmanifest.json";
+        function walk(resourcePath, internalPath) {
+            var list = fs.readdirSync(path.join(resourcePath, internalPath));
+            list.forEach(function (file) {
+                var sourcePathRelative = path.join(internalPath, file);
+                var sourcePath = path.join(resourcePath, sourcePathRelative);
+                var stat = fs.statSync(sourcePath);
+                if (stat && stat.isDirectory()) {
+                    walk(resourcePath, sourcePathRelative);
+                } else {
+                    if (file.match(/\.json$/)) {
+                        resourceList.push(sourcePathRelative);
+                    }
+                }
+            });
+        }
+
+        walk(resourcePath, "");
+        var manifest = {};
+        manifest.files = resourceList;
+        fs.writeFileSync(path.join(resourcePath, manifestName), JSON.stringify(manifest), 'utf8');
+    }
```

Comment:
Probably all of this code should be in the JsonResourceFileType class in the webos plugin, as it is specific to webos/ilib. Also, different file types can have different resource directories. For example, if you had an app with some java in the backend and some javascript in the front end, you could put the java properties files in one resource directory and the json files in a different directory. That way, the build can put them in separate packages. Each may need its own type of manifest.